tests/e2e/e2e_permissive_test.go

```diff
@@ -0,0 +1,107 @@
+package e2e
+
+import (
+	"fmt"
+	"strings"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("Permissive Traffic Policy Mode", func() {
+	Context("PermissiveMode", func() {
+		const sourceNs = "client"
+		const destNs = "server"
+		var ns []string = []string{sourceNs, destNs}
+
+		It("Tests HTTP traffic for client pod -> server pod with permissive mode", func() {
+			// Install OSM
+			installOpts := td.GetOSMInstallOpts()
+			installOpts.enablePermissiveMode = true
+			Expect(td.InstallOSM(installOpts)).To(Succeed())
+			Expect(td.WaitForPodsRunningReady(td.osmNamespace, 90*time.Second, 1)).To(Succeed())
+
+			// Create Test NS
+			for _, n := range ns {
+				Expect(td.CreateNs(n, nil)).To(Succeed())
+				Expect(td.AddNsToMesh(true, n)).To(Succeed())
+			}
+
+			// Get simple pod definitions for the HTTP server
+			svcAccDef, podDef, svcDef := td.SimplePodApp(
+				SimplePodAppDef{
+					name:      "server",
+					namespace: destNs,
+					image:     "kennethreitz/httpbin",
+					ports:     []int{80},
+				})
+
+			_, err := td.CreateServiceAccount(destNs, &svcAccDef)
+			Expect(err).NotTo(HaveOccurred())
+			dstPod, err := td.CreatePod(destNs, podDef)
+			Expect(err).NotTo(HaveOccurred())
+			_, err = td.CreateService(destNs, svcDef)
+			Expect(err).NotTo(HaveOccurred())
+
+			Expect(td.WaitForPodsRunningReady(destNs, 90*time.Second, 1)).To(Succeed())
+
+			// Get simple Pod definitions for the client
+			svcAccDef, podDef, svcDef = td.SimplePodApp(SimplePodAppDef{
+				name:      "client",
+				namespace: sourceNs,
+				command:   []string{"/bin/bash", "-c", "--"},
+				args:      []string{"while true; do sleep 30; done;"},
+				image:     "songrgg/alpine-debug",
+				ports:     []int{80},
+			})
+
+			_, err = td.CreateServiceAccount(sourceNs, &svcAccDef)
+			Expect(err).NotTo(HaveOccurred())
+			srcPod, err := td.CreatePod(sourceNs, podDef)
+			Expect(err).NotTo(HaveOccurred())
+			_, err = td.CreateService(sourceNs, svcDef)
+			Expect(err).NotTo(HaveOccurred())
+
+			Expect(td.WaitForPodsRunningReady(sourceNs, 90*time.Second, 1)).To(Succeed())
+
+			req := HTTPRequestDef{
```

Comment:
personal POV:

I'd add more complexity (maybe we can do it on another test? but it will be a superset of this; but I'd use multiple deployments and multiple pods per svc, etc). 
Permissive pushes everything everywhere, I'm just thinking of potential issues that would not show up until we put some complexity on the mesh itself.

That's not to say this test is not good, but if we have a larger test that covers more, I wonder if we'll really need a smaller one at that point (in terms of timing and number of tests, etc).

Reply:
Yeah I agree. I like the idea of still keeping the smallest possible example and running a more complex test in addition. That could help us better pinpoint problems as they arise.

Reply:
+1, agreed having smaller test too.